addons/maintenance/models/maintenance.py

```diff
@@ -176,7 +176,7 @@ def _compute_next_maintenance(self):
                 if next_date < date_now:
                     next_date = date_now
             else:
-                next_date = equipment.effective_date + timedelta(days=equipment.period)
+                next_date = equipment.effective_date + timedelta(days=equipment.period) if equipment.effective_date else False
```

Comment:
Maybe we want
```suggestion
next_date = equipment.effective_date + timedelta(days=equipment.period) if equipment.effective_date elsetimedelta(days=equipment.period)
```

I'm not sure, that's not my scope, what do you think about it ? @svs-odoo